commec/tools/blast_tools.py

```diff
@@ -110,6 +94,14 @@ def taxdist(blast, reg_ids, vax_ids, db_path, threads):
     reg = list(map(str, reg_ids[0]))
     vax = list(map(str, vax_ids[0]))
 
+    try:
+        a = t["FullLineage"].str.split(";")[0]
+        b = t["FullLineageTaxIDs"].str.split(";")[0]
+        c = t["FullLineageRanks"].str.split(";")[0]
+    except AttributeError:
+        logging.error("The Blast database used has not returned any Lineage information!")
+        return blast
```

Comment:
What's this test?

Reply:
From conversation: this prevents an error from being thrown if the BLAST database has no taxonomy information. Having thought about this more, we indeed shouldn't fail silently here, nor with a random AttributeError... what if we do something like:

```
    reg = list(map(str, reg_ids[0]))
    vax = list(map(str, vax_ids[0]))

    # checks which individual lines contain regulated pathogens
    lineage_info = pytaxonkit.lineage(blast["subject tax ids"], data_dir=db_path, threads=threads)
    if not all(col in df.columns for col in ["FullLineage", "FullLineageTaxIDs", "FullLineageRanks"]):
        logging.error(f"ERROR: Lineage information not returned based on NCBI taxonomy in: {db_path}")
        return blast
```

Reply:
Unfortunately the columns are present, just filled with none string objects, so this approach doesn't work.

Reply:
Just discussed a slight rephrase:
```
# Check that the required lineage information is present; same access as below. Can't just check that columns are present, since if db missing the data, they are present with null values
for required_lineage_col in ["FullLineage", "FullLineageTaxIDs", "FullLineageRanks"]:
    t[required_lineage_col].str.split(";")[0]
```

and then probably actually raise an exception?